luigi/scheduler.py

```diff
@@ -1312,6 +1077,14 @@ def get_work(self, host=None, assistant=False, current_tasks=None, worker=None,
         else:
             reply['task_id'] = None
 
+        # every 1000ish calls, make sure we sync the DB with our state in memory
+        if self._config.use_sql_state and int(time.time() * 1000) % 100 == 0:
```

Comment:
i wonder if this makes sense in the sql store itself, rather than high up here

Reply:
Hmm, that's a good point -- I'll look at this now, what it'd look like to move it further down

Reply:
donezo, how's it look?